src/authorization/WebAclReader.ts

```diff
@@ -56,17 +57,43 @@ export class WebAclReader extends PermissionReader {
    * Will throw an error if this is not the case.
    * @param input - Relevant data needed to check if access can be granted.
    */
-  public async handle({ identifier, credentials }: PermissionReaderInput):
+  public async handle({ identifier, credentials, modes }: PermissionReaderInput):
   Promise<PermissionSet> {
     // Determine the required access modes
     this.logger.debug(`Retrieving permissions of ${credentials.agent?.webId} for ${identifier.path}`);
 
     const isAcl = this.aclStrategy.isAuxiliaryIdentifier(identifier);
     const mainIdentifier = isAcl ? this.aclStrategy.getSubjectIdentifier(identifier) : identifier;
+    const checkParent = modes.has(AccessMode.create) || modes.has(AccessMode.delete);
 
-    // Determine the full authorization for the agent granted by the applicable ACL
-    const acl = await this.getAclRecursive(mainIdentifier);
-    return this.createPermissions(credentials, acl, isAcl);
+    // Determine the full authorization for the agent granted by the applicable ACL.
+    // Note that we don't filter on input modes as all results are needed for the WAC-Allow header.
```

Comment:
```suggestion
    // Rather than restricting the search to only the required modes,
    // we collect all modes in order to have complete metadata (for instance, for the WAC-Allow header).
```